File: sample-example.go
package main

import (
	"fmt"
	"io"
	"log"
	"os"

	"github.com/sahilm/ffmpeg-go"
)

func main() {
	inputFile, err := os.Open("input.mp4")
	if err != nil {
		log.Fatal(err)
	}
	defer inputFile.Close()

	outputFile, err := os.Create("output.mp4")
	if err != nil {
		log.Fatal(err)
	}
	defer outputFile.Close()

	decoder, err := ffmpeg.NewDecoder(inputFile)
	if err != nil {
		log.Fatal(err)
	}
	defer decoder.Close()

	encoder, err := ffmpeg.NewEncoder(outputFile)
	if err != nil {
		log.Fatal(err)
	}
	defer encoder.Close()

	for {
		packet, err := decoder.Decode()
		if err == io.EOF {
			break
		} else if err != nil {
			log.Fatal(err)
		}

		err = encoder.Encode(packet)
		if err != nil {
			log.Fatal(err)
		}
	}

	fmt.Println("Finished transcoding")
}

File: sample-example.ts
import { FfmpegStreamer } from "ffmpeg-streamer";

const inputFile = "input.mp4";
const outputFile = "output.mp4";

const streamer = new FfmpegStreamer({
  input: inputFile,
  output: outputFile,
});

streamer.on("start", () => {
  console.log("Started transcoding");
});

streamer.on("progress", (progress) => {
  console.log(`Transcoding progress: ${progress}%`);
});

streamer.on("end", () => {
  console.log("Finished transcoding");
});

streamer.on("error", (error) => {
  console.error(`Error transcoding: ${error}`);
});

streamer.start();
